plugins/search-backend-module-catalog/src/collators/DefaultCatalogCollatorFactory.ts

```diff
@@ -127,7 +126,7 @@ export class DefaultCatalogCollatorFactory implements DocumentCollatorFactory {
       catalogClient || new CatalogClient({ discoveryApi: discovery });
     this.entityTransformer =
       entityTransformer ?? defaultCatalogCollatorEntityTransformer;
-    this.auth = auth;
+    this.auth = createLegacyAuthAdapters(options).auth;
```

Comment:
note that this is already being done in fromConfig above

Reply:
Thank you for pointing that out, I dropped this change.